lib/elixir/lib/system.ex

```diff
@@ -312,9 +312,14 @@ defmodule System do
   """
   @spec user_home() :: String.t() | nil
   def user_home do
-    {:ok, [[home] | _]} = :init.get_argument(:home)
-    encoding = :file.native_name_encoding()
-    :unicode.characters_to_binary(home, encoding, encoding)
+    case :init.get_argument(:home) do
+      {:ok, [[home] | _]} ->
+        encoding = :file.native_name_encoding()
+        :unicode.characters_to_binary(home, encoding, encoding)
+
+      _ ->
+        nil
+    end
```

Comment:
Dialyzer suggests that `user_home` can never return `nil` and I agree. I am not sure if this is the way to fix this and I don't know how to test this :S

Reply:
Let's fix the typespecs for now to remove `| nil`.

Reply:
Doesn't seem right to me, there's a bang version underneath that relies on this returning a nil.

Reply:
This looks like the right way to fix this, per http://erlang.org/doc/man/init.html `get_name` returns `:error` when no value is associated with the flag.